### PR TITLE
Dasd 8817 migrate generate config task

### PR DESCRIPTION
**Pull request by NickGraham101**

### CODE REVIEW

File: Tests/IT.New-ConfigurationTableEntry.Helpers.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,60 @@
+Describe "New-ConfigurationTableEntry.Helper Integration Tests" -Tags @("Integration") {
+
+    BeforeAll {
+        Set-Location $PSScriptRoot/../Infrastructure-Scripts/New-ConfigurationTableEntry/
+        Import-Module ./tools/Helpers.psm1 -Force
+    }
+
+    Context "A valid schema and complete set of environment variables exists" {
+        if (!$ENV:TF_BUILD) {
+            Import-Module powershell-yaml
```
Comment (adam230594) [inline Tests/IT.New-ConfigurationTableEntry.Helpers.Tests.ps1:10]: Is a comment required to say that this module requires installing beforehand for visibility?

**Review by nbowes24 — DISMISSED** [review]

File: Infrastructure-Scripts/New-ConfigurationTableEntry/New-ConfigurationTableEntry.ps1 [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    [String]$StorageAccountName,
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [String]$StorageAccountResourceGroup,
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [String]$EnvironmentName,
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [String]$TableName,
```
[the remark this answers is not in the document]
Comment (NickGraham101) [reply Infrastructure-Scripts/New-ConfigurationTableEntry/New-ConfigurationTableEntry.ps1:62]: Corrected description

File: Infrastructure-Scripts/New-ConfigurationTableEntry/tools/Helpers.psm1 [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+    $SchemaDefinition = Get-Content -Path $SchemaDefinitionPath -Raw
+    $SchemaObject = [Newtonsoft.Json.Schema.JSchema, Newtonsoft.Json.Schema, Version = 2.0.0.0, Culture = neutral, PublicKeyToken = 30ad4fe6b2a6aeed]::Parse($SchemaDefinition)
+
+    Write-Host "Processing properties"
+    $Settings = [Newtonsoft.Json.JsonSerializerSettings, Newtonsoft.Json, Version = 9.0.0.0, Culture = neutral, PublicKeyToken = 30ad4fe6b2a6aeed]::new()
+    $Settings.MaxDepth = 100
+    $Schema = Expand-Schema -PropertyObject $SchemaObject.Properties
+    $Configuration = ($Schema | ConvertTo-Json -Depth 10 -Compress)
+
+    if ($PSVersionTable.PSVersion.Major -lt 6) {
```
[the remark this answers is not in the document]
Comment (NickGraham101) [reply Infrastructure-Scripts/New-ConfigurationTableEntry/tools/Helpers.psm1:45]: There's a follow up ticket for improvements, will add to that

File: Tests/IT.New-ConfigurationTableEntry.Helpers.Tests.ps1 (hunk 1 above)
Comment (NickGraham101) [reply Tests/IT.New-ConfigurationTableEntry.Helpers.Tests.ps1:11]: done

**Review by nbowes24 — APPROVED** [review]

**Pull request merged**